dashboard/app/controllers/api/v1/section_libraries_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Api::V1::SectionLibrariesController < Api::V1::JsonApiController
+  check_authorization
+
+  # gets the libraries from all members of all sections I am a part of
+  # GET api/v1/section_libraries
+  def index
+    # user is logged in
+    authorize! :manage, current_user
```

Comment:
this is a clever way to check for login and your comments make it clear what you're doing, but it's a bit more roundabout than it needs to be since it relies on the assumption that a signed-in user has certain abilities. Please add `before_action :authenticate_user!` to the class instead, for example like we do here: https://github.com/code-dot-org/code-dot-org/blob/77dc975a4253c3d054b8151e65b435fbdb7d616a/dashboard/app/controllers/api/v1/user_scripts_controller.rb#L1-L2